### PR TITLE
tapchannel: fix split root when initiator has 0 balance

### PR DESCRIPTION
**Pull request by guggero**
Fixes the last item described in https://github.com/lightninglabs/taproot-assets/issues/1019.

Validation integration test here: https://github.com/lightninglabs/lightning-terminal/pull/796.

This fixes a bug that lead to a panic in the following edge case:
 - The initiator doesn't have any asset balance (everything was moved to the responder).
 - An HTLC is created that sends some assets back to the initiator.

The problem that created is that because the initiator doesn't have any non-HTLC balance, no allocation was created with SplitRoot=true, which lead to a nil pointer panic in the logic that extracts the split root locator when creating a split commitment.

This commit fixes the problem by assigning the first HTLC that pays to the initiator with SplitRoot=true, if the initiator has zero balance.

### CODE REVIEW

**Review by Roasbeef — CHANGES_REQUESTED** [review]

File: tapchannel/commitment.go [hunk 1]
```diff
@@ -680,6 +724,7 @@ func CreateAllocations(chanState *channeldb.OpenChannel, ourBalance,
 			Type:           allocType,
 			Amount:         rfqmsg.Sum(htlc.AssetBalances),
 			AssetVersion:   asset.V1,
+			SplitRoot:      shouldHouseSplitRoot,
```
[the remark this answers is not in the document]
Comment (guggero) [reply tapchannel/commitment.go:727]: If the initiator has any asset (either as a channel balance or as an HTLC in their direction) then that allocation gets the split root (in case of multiple HTLCs, just the first). If there is no balance for the initiator (all settled towards the responder), then there is no split root and it's a full value send. No zero value tombstone needed because all outputs are marked as interactive.

File: tapchannel/commitment.go [hunk 2]
```diff
@@ -652,8 +662,12 @@ func CreateAllocations(chanState *channeldb.OpenChannel, ourBalance,
 			"allocations: %w", err)
 	}
 
+	log.Tracef("Channel initiator's asset balance is %d",
+		initiatorAssetBalance)
+
 	// Next, we add the HTLC outputs, using this helper function to
 	// distinguish between incoming and outgoing HTLCs.
+	var haveHtlcSplitRoot bool
```
Comment (ffranr) [inline tapchannel/commitment.go:670]: I think we should add a comment here to explain how `haveHtlcSplitRoot` is used and why it exists.

**Review by ffranr — APPROVED** [review]

**Review by Roasbeef — APPROVED** [review]
LGTM 🦜
